packages/@lightningjs/ui-components/src/components/Badge/Badge.d.ts

```diff
@@ -33,16 +33,81 @@ export type BadgeStyle = {
   textStyle: lng.textures.TextTexture.Settings;
 };
 
-export default class Badge extends Base {
-  title?: string;
+declare namespace Badge {
+  export interface TemplateSpec extends Base.TemplateSpec {
+    /**
+     * path to image or inline SVG XML
+     */
+    icon?: string;
+
+    /**
+     * width of icon
+     */
+    iconWidth?: number;
+
+    /**
+     * height of icon
+     */
+    iconHeight?: number;
+
+    /**
+     * side of the text where icon will appear on (`left` or `right`)
+     */
+    iconAlign?: 'left' | 'right';
+
+    /**
+     * Badge text
+     */
+    title?: string;
+  }
+
+  export interface TypeConfig extends lng.Component.TypeConfig {
+    SignalMapType: SignalMap;
+  }
+  /**
+   * emitted when the Badge component finishes loading its content and adjusts the background to fit the text input size.
+   */
+  export type SignalMap = {
+    loadedBadge(): void;
+  };
+}
+
+declare class Badge<
+  TemplateSpec extends Badge.TemplateSpec = Badge.TemplateSpec,
+  TypeConfig extends lng.Component.TypeConfig = lng.Component.TypeConfig
```

Comment:
this needs to be updated to extend Badge.TypeConfig instead of lng.Component.TypeConfig